File: 2025-2/sesion-de-clase-19/.net/SoftProgNegocio/BOImpl/ProductoBOImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PUCP.SoftProg.Modelo.Almacen;
using PUCP.SoftProg.Negocio.BO;
using PUCP.SoftProg.Persistencia.DAO.Almacen;
using PUCP.SoftProg.Persistencia.DAOImpl.Almacen;
using PUCP.SoftProg.Modelo;

namespace PUCP.SoftProg.Negocio.BOImpl {
    public class ProductoBOImpl : IProductoBO {
        private readonly IProductoDAO productoDAO;

        public ProductoBOImpl() {
            this.productoDAO = new ProductoDAOImpl();
        }

        public void Guardar(Producto producto, Estado estado) {
            if (estado == Estado.Nuevo) {
                this.productoDAO.Crear(producto);
            }
            else { 
                this.productoDAO.Actualizar(producto);
            }
        }

        public void Eliminar(int id) {
            this.productoDAO.Eliminar(id);
        }

        public Producto Obtener(int id) {
            return this.productoDAO.Leer(id);
        }

        public List<Producto> Listar() {
            return this.productoDAO.LeerTodos();
        }        
    }
}

File: 2025-2/sesion-de-clase-22/java/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/boimpl/almacen/ProductoBOImpl.java
package pe.edu.pucp.inf30.softprog.boimpl.almacen;

import java.util.List;
import pe.edu.pucp.inf30.softprog.bo.almacen.ProductoBO;
import pe.edu.pucp.inf30.softprog.modelo.Estado;
import pe.edu.pucp.inf30.softprog.modelo.almacen.Producto;
import pe.edu.pucp.inf30.softprog.dao.almacen.ProductoDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.almacen.ProductoDAOImpl;

/**
 *
 * @author eric
 */
public class ProductoBOImpl implements ProductoBO {
    private final ProductoDAO productoDao;

    public ProductoBOImpl() {
        this.productoDao = new ProductoDAOImpl();
    }
    
    @Override
    public List<Producto> listar() {
        return this.productoDao.leerTodos();
    }

    @Override
    public Producto obtener(int id) {
        return this.productoDao.leer(id);
    }

    @Override
    public void eliminar(int id) {
        this.productoDao.eliminar(id);
    }

    @Override
    public void guardar(Producto modelo, Estado estado) {
        if (estado == Estado.Nuevo) {
            this.productoDao.crear(modelo);
        }
        else {
            this.productoDao.actualizar(modelo);
        }
    }
}
